modules/documents/client/directives/documents.confirm.delete.directive.js

```diff
@@ -106,29 +106,27 @@ angular.module('documents')
 					function checkFoldersForContent() {
 						var promises = [];
 						//We check all folders (instead of just deletable folders) since a folder can be published and can have children
-                        			//In that case, we need to display two warnings
+						//In that case, we need to display two warnings
 						_.forEach(self.folders, function(fldr) {
 							// Does the folder have child folders? ....
 							var node = self.currentNode.first(function (child) {
 								return (child.model.id === fldr.model.id);
 							});
-							fldr.hasChildren = node.hasChildren();
-							fldr.canBeDeleted = !fldr.hasChildren;
-							if (fldr.canBeDeleted) {
-								// Does the folder have document content?....
-								promises.push(new Promise (function (resolve, reject) {
-									DocumentMgrService.getDirectoryDocuments(self.project, fldr.model.id)
-									.then(function (result) {
-										fldr.hasChildren = result.data.length > 0;
-										fldr.canBeDeleted = !fldr.hasChildren;
-										return resolve(fldr);
-									})
-									.catch(function (err) {
-										console.log("Error", err);
-										return reject(err);
-									});
-								}));
-							}
+							//Does the folder have document content?....
+							promises.push(new Promise (function (resolve, reject) {
+								DocumentMgrService.getDirectoryDocuments(self.project, fldr.model.id)
+								.then(function (result) {
+									//A folder could have child folders (nodes) or could have files (end nodes)
+									fldr.hasChildren = node.hasChildren() || result.data.length > 0 ;
+									// If a folder has children or has been published, it cannot be deleted.
+									fldr.canBeDeleted = !(fldr.hasChildren || fldr.isPublished);
+									return resolve(fldr);
+								})
+								.catch(function (err) {
+									console.log("Error", err);
```

Comment:
Remove console.log, we shouldn't be dumping these in prod.

Reply:
done